pkg/cluster/px/px.go

```diff
@@ -524,21 +527,27 @@ func (ops *pxClusterOps) upgradePX(newVersion string) error {
 			return nil, true, errNoPXDaemonset
 		}
 
+		pxImage := fmt.Sprintf("%s:%s", spec.PXImage, spec.PXTag)
+		ociImage := fmt.Sprintf("%s:%s", spec.OCIMonImage, spec.OCIMonTag)
+
 		for _, ds := range dss {
 			skip := false
 			logrus.Infof("Upgrading PX daemonset: [%s] %s to version: %s", ds.Namespace, ds.Name, newVersion)
 			dsCopy := ds.DeepCopy()
 			for i := 0; i < len(dsCopy.Spec.Template.Spec.Containers); i++ {
 				c := &dsCopy.Spec.Template.Spec.Containers[i]
 				if c.Name == pxDaemonsetContainerName {
-					if c.Image == newVersion {
+					if c.Image == ociImage && spec.OCIMonTag == spec.PXTag {
```

Comment:
you also need to check here if PX_IMAGE env variable exists in current container that points to a different value than pxImage that user gave for upgrade.